### ISSUE TITLE
Trivy client server mode not scanning secrets exposed in image, Trivy standalone works.

### ISSUE THREAD

**Issue by gsingh737** [aquasecurity/trivy-operator#1836]
**What steps did you take and what happened:**
Running Trivy operator with these ENV values
```
OPERATOR_TARGET_NAMESPACES:
OPERATOR_EXCLUDE_NAMESPACES:
OPERATOR_TARGET_WORKLOADS: pod,replicaset,replicationcontroller,statefulset,daemonset,cronjob,job
OPERATOR_SERVICE_ACCOUNT: trivy-operator
OPERATOR_LOG_DEV_MODE: false
OPERATOR_SCAN_JOB_TTL:
OPERATOR_SCAN_JOB_TIMEOUT: 5m
OPERATOR_CONCURRENT_SCAN_JOBS_LIMIT: 10
OPERATOR_CONCURRENT_NODE_COLLECTOR_LIMIT: 1
OPERATOR_SCAN_JOB_RETRY_AFTER: 30s
OPERATOR_BATCH_DELETE_LIMIT: 10
OPERATOR_BATCH_DELETE_DELAY: 10s
OPERATOR_METRICS_BIND_ADDRESS: :8080
OPERATOR_METRICS_FINDINGS_ENABLED: true
OPERATOR_METRICS_VULN_ID_ENABLED: false
OPERATOR_HEALTH_PROBE_BIND_ADDRESS: :9090
OPERATOR_VULNERABILITY_SCANNER_ENABLED: true
OPERATOR_SBOM_GENERATION_ENABLED: true
OPERATOR_CLUSTER_SBOM_CACHE_ENABLED: false
OPERATOR_VULNERABILITY_SCANNER_SCAN_ONLY_CURRENT_REVISIONS: true
OPERATOR_SCANNER_REPORT_TTL: 24h
OPERATOR_CACHE_REPORT_TTL: 120h
CONTROLLER_CACHE_SYNC_TIMEOUT: 5m
OPERATOR_CONFIG_AUDIT_SCANNER_ENABLED: true
OPERATOR_RBAC_ASSESSMENT_SCANNER_ENABLED: true
OPERATOR_INFRA_ASSESSMENT_SCANNER_ENABLED: true
OPERATOR_CONFIG_AUDIT_SCANNER_SCAN_ONLY_CURRENT_REVISIONS: true
OPERATOR_EXPOSED_SECRET_SCANNER_ENABLED: true
OPERATOR_METRICS_EXPOSED_SECRET_INFO_ENABLED: false
OPERATOR_METRICS_CONFIG_AUDIT_INFO_ENABLED: false
OPERATOR_METRICS_RBAC_ASSESSMENT_INFO_ENABLED: false
OPERATOR_METRICS_INFRA_ASSESSMENT_INFO_ENABLED: false
OPERATOR_METRICS_IMAGE_INFO_ENABLED: false
OPERATOR_METRICS_CLUSTER_COMPLIANCE_INFO_ENABLED: false
OPERATOR_WEBHOOK_BROADCAST_URL:
OPERATOR_WEBHOOK_BROADCAST_TIMEOUT: 30s
OPERATOR_SEND_DELETED_REPORTS: false
OPERATOR_PRIVATE_REGISTRY_SCAN_SECRETS_NAMES: {}
OPERATOR_ACCESS_GLOBAL_SECRETS_SERVICE_ACCOUNTS: true
OPERATOR_BUILT_IN_TRIVY_SERVER: true
TRIVY_SERVER_HEALTH_CHECK_CACHE_EXPIRATION: 10h
OPERATOR_MERGE_RBAC_FINDING_WITH_CONFIG_AUDIT: false
OPERATOR_CLUSTER_COMPLIANCE_ENABLED: true

```

Built a Dockerfile with following 
```
FROM nginx:latest

ENV AWS_SECRET_ACCESS_KEY hKTiTyckWru9Bji4OCm7Keif7fm7Uz11LmTjbx

CMD ["nginx", "-g", "daemon off;"]

```

Running a pod in cluster with Trivy Operator running with Trivy server in Client/Server Mode
![image](https://github.com/aquasecurity/trivy-operator/assets/11049175/45b27555-e9b3-4e4c-81a2-fcdb26a63e58)

Exposed Secret Report is not catching any secret. 
![image](https://github.com/aquasecurity/trivy-operator/assets/11049175/ee3d76e0-c917-48e8-88ad-ddbf5a2084d9)

[A clear and concise description of what the bug is, and what commands you ran.]

**What did you expect to happen:**
Expected Trivy operator to catch exposed secrets.

**Anything else you would like to add:**
Running in standalone Trivy image scan  catches exposed secrets using Trivy cli i.e
`trivy image  --image-config-scanners secret --scanners secret --timeout 10m nginxwithsecret:v1`

![image](https://github.com/aquasecurity/trivy-operator/assets/11049175/956fc3b4-d5e6-4a25-94b9-7bb6eeb8f83d)

When running Trivy cli using Trivy Server, its back to not catching those secrets for the same image.
 `trivy image --server http://localhost:4954 --image-config-scanners secret --scanners secret --timeout 10m nginxwithsecret:v1`
![image](https://github.com/aquasecurity/trivy-operator/assets/11049175/6be2f291-18d0-4e8a-b18b-282dd8866c65)
![image](https://github.com/aquasecurity/trivy-operator/assets/11049175/74c1e110-43c4-4f57-b2f0-03c466c29b6f)

This issue was also reported here 
https://github.com/aquasecurity/trivy-operator/issues/1297 and was thought to be fixed in https://github.com/aquasecurity/trivy-operator/pull/1301
But I believe the fix is addressing FS mode scanning not image scanning as fix was only put in 
`func (p *plugin) getFSScanningArgs(ctx trivyoperator.PluginContext, command Command, mode Mode, trivyServerURL string) []string `

**Environment:**

- Trivy-Operator version (use `trivy-operator version`): 0.18.3  Chart version: 0.20.4
- Kubernetes version (use `kubectl version`): v1.27.7

**Comment by chen-keinan** [comment]
@gsingh737 thanks for the input, I'll have a look. 
 btw: do you have a public image name where I can use it for testing ?

**Comment by gsingh737** [comment]
@chen-keinan gsingh737/nginxwithsecret:v1 its public on dockerhub.

**Comment by chen-keinan** [comment]
@gsingh737 sorry for getting late to this, I have tried scanning you image with trivy 0.51.1, image mode and did not got any results:
```sh
trivy image gsingh737/nginxwithsecret:v1 --image-config-scanners secret --scanners secret
2024-05-20T11:55:30+03:00	INFO	Container image config scanners	scanners=[secret]
2024-05-20T11:55:30+03:00	INFO	Secret scanning is enabled
2024-05-20T11:55:30+03:00	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-05-20T11:55:30+03:00	INFO	Please see also https://aquasecurity.github.io/trivy/v0.51/docs/scanner/secret/#recommendation for faster secret detection
```
am I missing anything ?

**Comment by jemag** [comment]
Weird, here is me trying it with 0.48.0
![image](https://github.com/aquasecurity/trivy-operator/assets/7985687/dffe3427-b86a-4851-a88e-96bb4624d55c)
and then with 0.51.2:
![image](https://github.com/aquasecurity/trivy-operator/assets/7985687/ca1412e3-9ebf-41d9-ab8d-db2e47c58284)

In both case it is reported

**Comment by chen-keinan** [comment]
@jemag does it works for you in client/server mode:
```trivy server --listen 0.0.0.0:8081```

AND

```trivy image gsingh737/nginxwithsecret:v1 --image-config-scanners secret --scanners secret --server http://0.0.0.0:8081```

**Comment by chen-keinan** [comment]
Related https://github.com/aquasecurity/trivy/issues/6742

**Comment by jemag** [comment]
I do detect the image secret with regular trivy-cli:
![image](https://github.com/aquasecurity/trivy-operator/assets/7985687/d685cd4d-0b53-415e-8392-cdda4701212f)
However I do not detect it running the trivy-server included in trivy-operator (port-forwarding it locally):
![image](https://github.com/aquasecurity/trivy-operator/assets/7985687/59415c67-317b-4607-8a42-e503d07f7c04)

**Comment by chen-keinan** [comment]
@jemag , thanks for checking it out, I have made several tests and got mixed result.
I'll investigate it

**Comment by dklimpel** [comment]
I seem to have the same problem. Is there any chance of following up or supporting this issue?

**Comment by afdesk** [comment]
@dklimpel thanks for the report
we're on it right now

**Comment by dklimpel** [comment]
I have looked again. In principle, `ExposedSecretReport` is created (also in client/server mode).

- If I delete a `VulnerabilityReport`, it is automatically recreated.
- If I delete an `ExposedSecretReport`, it is not automatically recreated. 


`ExposedSecretReport` is created if I first delete `ClusterSbomReport` and then delete `ExposedSecretReport`. Or if I delete `ExposedSecretReport`, then `ClusterSbomReport` and then additionally the `VulnerabilityReport`.

Maybe that helps.

My configuration:

```yaml
serviceMonitor:
  enabled: true

excludeNamespaces: "kube-system"

operator:
  builtInTrivyServer: true
  clusterSbomCacheEnabled: true
  metricsExposedSecretInfo: true
  metricsImageInfo: true
  metricsVulnIdEnabled: true

trivyOperator:
  scanJobCustomVolumesMount:
    - name: secret-config
      mountPath: /trivy-secret.yaml
      subPath: trivy-secret.yaml
      readOnly: true

  scanJobCustomVolumes:
    - name: secret-config
      secret:
        secretName: trivy-custom-secret-rules

service:
  headless: false
```

**Comment by afdesk** [comment]
https://github.com/aquasecurity/trivy-operator/pull/2590 was reverted, so let's re-open it

**Comment by afdesk** [comment]
@dklimpel thanks for the details
I spent some time to investigate `client-server` mode based on your comment.

It seems there is another issue. I see you use a custom secret config, may be there is something there.
just note: by design `secret` scanner always works on client side.

in any case, after #2595 the both modes (standalone and client-server) should have the same behavior and work correctly.

**Comment by dklimpel** [comment]
@afdesk Ok, thank you.

I will retest when #2595 has landed.
I can work with the workaround.
In my point of view you can close this issue if the origin case is fixed.

**Comment by afdesk** [comment]
@dklimpel great! we'll wait for your feedback

but my main concern there is another hidden issue here - for example, a custom volume isn't mounted correctly...

**Comment by dklimpel** [comment]
> another hidden issue here

I think not. If the scan runs, the scan shows results from the custom mapped config file.
